qualtrics_link/util.py

```diff
@@ -179,41 +184,131 @@ def is_user_in_whitelist(huid):
             return True
     except QualtricsAccessList.DoesNotExist:
         # the user is not in the whitelist'
+        logger.info('The HUID {} does not exist within the Qualtrics access list.'.format(huid))
         return False
 
 
-def build_user_dict(data):
+# If there are more than one record returned for a HUID,
+# we want to determine if any are employee status and to return that record.
+def filter_person_list(person_list):
+    for person in person_list:
+        if person.role_type_cd.lower() == 'employee':
+            return person
+    # If no employee records were found, then return the first person in the list
+    return person_list[0]
 
-    user_data = {
-        'role': 'generic',
-        'division': 'Other',
-        'validschool': False,
-        'validdept': False
+
+# Get the person list matching the given HUID or return None
+def get_person_list(huid):
+    person_list = Person.objects.filter(univ_id=huid, prime_role_indicator='Y')
+    return person_list
+
+
+# This function is waiting on a Qualtrics API update to be able to get a user by their username
+# Will update the given HUID users current role and division for their Qualtrics account
+def update_qualtrics_user(huid, division, role):
+    enc_id = get_encrypted_huid(huid).join('#harvard')
+    token = settings.QUALTRICS_LINK.get('QUALTRICS_API_TOKEN')
+    req_params = {
+        'divisionId': division,
+        'userType': role
     }
 
-    if 'people' in data:
-        person = data['people'][0]
-        
-        user_data['firstname'] = person.get('firstName', 'Not Available')
-        user_data['lastname'] = person.get('lastName', 'Not Available')
-        user_data['email'] = person.get('email', 'Not Available')
-
-        # School Affiliations check
-        school_affiliations = person.get('schoolAffiliations', 'Not Available')
-
-        valid_school_code = get_valid_school(school_affiliations)
-        if valid_school_code is not None:
-            user_data['validschool'] = True
-            user_data['role'] = 'student'
-            user_data['division'] = valid_school_code
-
-        # Department Affiliations check
-        department_affiliation = person.get('departmentAffiliation', 'Not Available')
-        if department_affiliation.lower() != 'not available':
-            valid_department = get_valid_dept(department_affiliation)
-            if valid_department is not None:
-                user_data['validdept'] = True
-                user_data['role'] = 'employee'
-                user_data['division'] = valid_department
-
-    return user_data
+    requests.put(url='https://harvard.qualtrics.com/API/v3/users/{}'.format(enc_id),
+                 data=req_params,
+                 headers={'X-API-TOKEN': token})
+
+# This function is waiting on a Qualtrics API update to be able to get a user by their username
+# Query Qualtrics to get the user with the given HUID
+def get_qualtrics_user(huid):
```

Comment:
Another nitpick: It would be be good to use the triple double quotes for docstrings - per PEP8 Style Guide and to  keep it consistent with most of our code. (very minor)